tests/integration/pilot/routing_test.go

```diff
@@ -309,96 +347,128 @@ func expectString(got, expected, help string) error {
 }
 
 // Todo merge with security TestReachability code
-func protocolSniffingCases() []TrafficTestCase {
+func protocolSniffingCases(ctx framework.TestContext) []TrafficTestCase {
+
 	cases := []TrafficTestCase{}
-	for _, client := range []echo.Instance{apps.podA, apps.naked, apps.headless} {
-		for _, destination := range []echo.Instance{apps.podA, apps.naked, apps.vmA, apps.headless} {
-			client := client
-			destination := destination
-			if client == apps.naked && destination == apps.vmA {
-				// Need a sidecar to connect to VMs
-				continue
+	// TODO add VMs to clients when DNS works for VMs.
+	for _, clients := range []echo.Instances{apps.podA, apps.naked, apps.headless} {
+		for _, client := range clients {
+
+			destinationSets := []echo.Instances{
+				apps.podA,
+				{apps.vmA},
+				apps.headless.Match(echo.InCluster(client.Config().Cluster)),
 			}
-			for _, call := range []struct {
-				// The port we call
-				port string
-				// The actual type of traffic we send to the port
-				scheme scheme.Instance
-			}{
-				{"http", scheme.HTTP},
-				{"auto-http", scheme.HTTP},
-				{"tcp", scheme.TCP},
-				{"auto-tcp", scheme.TCP},
-				{"grpc", scheme.GRPC},
-				{"auto-grpc", scheme.GRPC},
-			} {
-				cases = append(cases, TrafficTestCase{
-					name: fmt.Sprintf("%v %v %v", call.port, client.Config().Service, destination.Config().Service),
-					call: func() (echoclient.ParsedResponses, error) {
-						return client.Call(echo.CallOptions{Target: destination, PortName: call.port, Scheme: call.scheme})
-					},
-					validator: func(responses echoclient.ParsedResponses) error {
-						return responses.CheckOK()
-					},
-				})
+
+			if !ctx.Environment().IsMultinetwork() {
+				// TODO(#26517) fix multinetwork with non-sidecar pods, for now only test this in single-network
+				// the filter below assumes we fix this by restricting cross-network for non-sidecars
+				apps.naked.Match(echo.InNetwork(client.Config().Cluster.NetworkName()))
```

Comment:
isn't this doing nothing? I don't see where you do anything with the result

Reply:
fixed